src/pke/lib/encoding/ckkspackedencoding.cpp

```diff
@@ -374,6 +374,160 @@ bool CKKSPackedEncoding::Encode() {
 }
 #endif
 
+bool CKKSPackedEncoding::EncodeWithExtra() {
```

Comment:
@sarojaduality The logic for `EncodeWithExtra` seems to unnecessarily complicated.  For `Encode`, we only support DCRTPoly. We should support only DCRTPoly here, too. 

1. Please remove the two other branches from `EncodeWithExtra`. 
2. Please review the implementation of `Encode` (the logic should the same for `Encode` and `EncodeWithExtra`, except for the extra part). Maybe `EncodeWithExtra` can be combined with `Encode`.

Reply:
Removed the EncodeWithExtra function, so we only use Encode instead.